### PR TITLE
Add IMDB data and loaders

### PR DESCRIPTION
**Pull request by brs96**
Co-authored-by: Adam Schill Collberg <adam.schill.collberg@protonmail.com>

Thank you for your contribution to the Graph Data Science Client project.

Before submitting this PR, please read [Contributing to the Neo4j Ecosystem](https://github.com/neo4j/graph-data-science-client/blob/main/CONTRIBUTING.md). 

Make sure:
- [x] You signed the [Neo4j CLA](https://neo4j.com/developer/cla/#sign-cla) (Contributor License Agreement) so that we are allowed to ship your code in our library
- [x] Your contribution is covered by tests

### CODE REVIEW

**Comment by brs96** [comment]
[Adding additional pre-canned datasets. For example: 1 heterogeneous graph and 1 social network.](https://trello.com/c/VT6zBTSe/7304-adding-additional-pre-canned-datasets-for-example-1-heterogeneous-graph-and-1-social-network)

**Comment by brs96** [comment]
NO merging until it's confirmed that hosting datasets are fine.

**Review by FlorentinD — COMMENTED** [review]
Nice additions!

File: graphdatascience/resources/imdb/serialize_imdb.py
Comment (FlorentinD) [reply graphdatascience/resources/imdb/serialize_imdb.py:35]: Talking with Martin, `NaN` we expect to support.
Setting values to `null` we cannot, as what is the primitive version of `null` after all

File: graphdatascience/resources/imdb/serialize_imdb.py
Comment (brs96) [reply graphdatascience/resources/imdb/serialize_imdb.py:9]: https://github.com/seongjunyun/Graph_Transformer_Networks/tree/master/prev_GTN

File: graphdatascience/resources/imdb/serialize_imdb.py
Comment (brs96) [reply graphdatascience/resources/imdb/serialize_imdb.py:35]: Tried `np.nan` and `float("NaN")`, neither seems to work... Any other way of doing it?

File: graphdatascience/graph/graph_proc_runner.py
Comment (adamnsch) [reply graphdatascience/graph/graph_proc_runner.py:57]: Yeah I don't think we want to load it "undirected" at this point. Maybe we can add an optional parameter `orientation="NATURAL"` in the future when we have the support for it instead

File: graphdatascience/resources/imdb/serialize_imdb.py
Comment (adamnsch) [inline graphdatascience/resources/imdb/serialize_imdb.py:1]: This file should be pruned out of packaging in `MANIFEST.in`

Comment (adamnsch) [reply graphdatascience/resources/imdb/serialize_imdb.py:1]: The path to `serialize_cora.py` there also needs to be updated now

**Review by adamnsch — CHANGES_REQUESTED** [review]
Did you verify that this works with the new file paths? If you package the library, and then install it on a "fresh system", does it find the resource files?

See this line in `setup.py`:
```
package_data={"graphdatascience": ["py.typed", "resources/**/*.pkl"]},
```
I think it should still be fine.

Other than that, just have some smaller remarks.

Nice work!

File: graphdatascience/resources/imdb/serialize_imdb.py
Comment (adamnsch) [reply graphdatascience/resources/imdb/serialize_imdb.py:35]: I think it could be nice to somehow separate movies that have class labels, and those that have not. For example, in addition to the "Movie" label, they could have either "Labeled" or "Unlabeled" labels. This way we would be able to use the dataset in our NC pipeline (which one of the reasons we are doing this), and you don't really have to worry about null/nan features. Quite simply, only ["Movie", "Labeled"] nodes would have the class label.
This is similar to what we're doing with Cypher in our quality benchmarks on this dataset.

File: graphdatascience/resources/imdb/serialize_imdb.py
Comment (FlorentinD) [reply graphdatascience/resources/imdb/serialize_imdb.py:9]: mind dropping a comment in the code :)

File: graphdatascience/resources/imdb/serialize_imdb.py
Comment (FlorentinD) [reply graphdatascience/resources/imdb/serialize_imdb.py:35]: can you card this @brs96 , that np.nan() does not work with arrow? I think its worth to investigate more (but not on this PR as I agree with Adam)

File: graphdatascience/resources/imdb/serialize_imdb.py
Comment (adamnsch) [inline graphdatascience/resources/imdb/serialize_imdb.py:47]: I think "Director" and "Actor" are better names

File: graphdatascience/resources/imdb/serialize_imdb.py
Comment (adamnsch) [inline graphdatascience/resources/imdb/serialize_imdb.py:18]: Might be call the property "genre" instead of "class" to give it some semantic meaning. And indeed, it may not be used for classification after all, in which "class" does not really make sense

Comment (adamnsch) [reply graphdatascience/resources/imdb/serialize_imdb.py:18]: The same comment for "feature" below. I would instead call it "plot_keywords" or something similar

**Review by adamnsch — COMMENTED** [review]
I don't think we should commit the "raw" data. It's very large for a github repo. Let's upload it to our google drive or something for safe keeping?

**Comment by FlorentinD** [comment]
i would suggest our s3 bucket

File: graphdatascience/tests/integration/test_database_ops.py
Comment (brs96) [reply graphdatascience/tests/integration/test_database_ops.py:42]: That's true, but db creation is before the retry logic and I think we probably don't cover DB creation failing here.

File: graphdatascience/resources/imdb/serialize_imdb.py
Comment (brs96) [reply graphdatascience/resources/imdb/serialize_imdb.py:1]: 👍  didn't know about this, that's good that we keep the package clean. Added now.

**Comment by brs96** [comment]
What's left I think are:
1. Rename some fields for imdb (and then recreated the pickles). Perhaps also cleanup 'class' = -1 for 'Person' nodes by either set them to NaN or allowing multiple nodeDFs in graph.construct for community users?
2. Move the 'raw 'files to s3.

File: graphdatascience/resources/imdb/serialize_imdb.py
Comment (adamnsch) [reply graphdatascience/resources/imdb/serialize_imdb.py:36]: I think we still need NaN support for this since since Cypher loading does not support using several data frames. So they must all have the label property

Comment (FlorentinD) [reply graphdatascience/resources/imdb/serialize_imdb.py:36]: NaN support was expected to work and we will look into why the python NaN does not work

**Comment by FlorentinD** [comment]
I just merged the null support for float values in graph.construct so this should unblock you from using `NaN` in the input data for 2.3+

**Comment by Mats-SX** [comment]
This is blocked by https://github.com/neo4j/graph-data-science-client/pull/225

**Comment by FlorentinD** [comment]
Are you referring to supporting multiple DFs in CE? This is not the idea of this PR but probably a follow-up.
Couldnt we merge the multiple DFs into one for this loader here?

**Comment by adamnsch** [comment]
> Are you referring to supporting multiple DFs in CE? This is not the idea of this PR but probably a follow-up. Couldnt we merge the multiple DFs into one for this loader here?

Ah ok. I don't think we should merge DFs here as we will just have to do unnecessary work. I'd rather wait :)

File: graphdatascience/tests/integration/test_database_ops.py
Comment (Mats-SX) [inline graphdatascience/tests/integration/test_database_ops.py:16]: It is possible to use blocking variants of Neo4j system commands. Usually with the syntax `WAIT` (and `NO WAIT`).

So doing `CREATE DATABASE foo WAIT` is a better alternative here.

**Comment by Mats-SX** [comment]
Extracted karate club stuff to #234.
Rebased this PR on top of it.

File: graphdatascience/resources/imdb/serialize_imdb.py
Comment (Mats-SX) [reply graphdatascience/resources/imdb/serialize_imdb.py:48]: How about using those as node labels, and using `DIRECTED` and `ACTED_IN` instead?

**Pull request merged**
